src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/MicrosoftCodeQualityAnalyzersResources.resx

```diff
@@ -1311,4 +1311,12 @@
   <data name="OperatorOverloadsHaveNamedAlternatesCodeFixTitle" xml:space="preserve">
     <value>Add operator overload named alternate</value>
   </data>
-</root>
+  <data name="AvoidMultipleEnumerationsMessage" xml:space="preserve">
+    <value>Whenever an 'IEnumerable' collection is used, it will be enumerated. Try converting it to other collections.</value>
```

Comment:
I think we should get PM opinion on a good message to use here. Tagging @mikadumont @gewarren

Reply:
Can you clarify what the analyzer is flagging? I thought IEnumerable collections only yielded as much data as was requested. Don't most collections implement IEnumerable? So I'm not sure what other collections are being proposed.

Reply:
@gewarren 
This analyzer tries to find this pattern.
```
var i = Enumerable.Range(1, 10);

var t1 = i.ElementAt(5);
var t2 = i.ElementAt(10);
```
t1, and t2 would cause `i` to be enumerated twice.
But if you convert `i` to an array/list/...etc beforehand then `i` would be enumerated only once.

Reply:
Thanks @Cosifne. In that case, how about:

To prevent unnecessary enumerations of the 'IEnumerable' collection, consider caching it first by calling 'ToList' or 'ToArray'.

Reply:
@gewarren 
I am not sure if the 'ToList' or 'ToArray' call is precise enough here because there are a lot of other choices. (Set, Dictionary, ImmmutableArray, etc.)
Is it a fine pattern to give the user the simplest solution (list ToList and ToArray) in the diagnostic?

Reply:
How about:

To prevent unnecessary enumerations of the 'IEnumerable' collection, consider caching it first by calling a method such as 'ToList' or 'ToArray'.